tools/make/lint.mk

```diff
@@ -4,27 +4,56 @@
 
 ##@ Lint
 
+GITHUB_ACTION ?=
+
 .PHONY: lint
 lint: ## Run all linter of code sources, including golint, yamllint, whitenoise lint and codespell.
-	@$(MAKE) lint.golint lint.yamllint lint.codespell lint.whitenoise
 
+PHONY: lint-deps
+lint-deps: ## Everything necessary to lint (useful to separate out in the logs)
+
+GOLANGCI_LINT_FLAGS ?= $(if $(GITHUB_ACTION),--out-format=github-actions)
 .PHONY: lint.golint
-lint.golint:
+lint: lint.golint
+lint-deps: $(tools/golangci-lint)
+lint.golint: $(tools/golangci-lint)
 	@echo Running Go linter ...
-	@golangci-lint run --build-tags=e2e --config=tools/linter/golangci-lint/.golangci.yml
+	$(tools/golangci-lint) run $(GOLANGCI_LINT_FLAGS) --build-tags=e2e --config=tools/linter/golangci-lint/.golangci.yml
 
 .PHONY: lint.yamllint
-lint.yamllint:
+lint: lint.yamllint
+lint-deps: $(tools/yamllint)
+lint.yamllint: $(tools/yamllint)
 	@echo Running YAML linter ...
-	@yamllint --config-file=tools/linter/yamllint/.yamllint changelogs/
+	$(tools/yamllint) --config-file=tools/linter/yamllint/.yamllint changelogs/
 
+CODESPELL_FLAGS ?= $(if $(GITHUB_ACTION),--disable-colors)
 .PHONY: lint.codespell
+lint: lint.codespell
```

Comment:
I find myself moving my head up and down the screen trying to compute the dependency list.
prefer the current way with the add on dependancy of installing the tool

Reply:
I don't have a real strong opinion here, but to make an argument for this way:
- Avoiding merge conflicts when 2 different people add 2 different linters
- When adding a linter, there are fewer places where it needs updated. Don't need to define the target in one place, add a `lint` dependency in a second place, and a `lint-deps` dependency in a third place. I don't think we should be adding bunches of linters, but as the list grows, this will be a bigger deal.

Reply:
this is purely a taste aspect, prefer the existing approach to simplify code maintenance but this comment is non blocking